### PR TITLE
Bump kubernetes client version to 1.17

### LINKED ISSUE #594: Bump Kubernetes client version for import-secrets
## Task

Bump up the Kubernetes client version for import-secrets, ideally to 1.17

https://github.com/openfaas/openfaas-cloud/tree/master/import-secrets

Good first issue - feel free to ask questions. e2e testing will be required with any PR and don't convert to Go modules at this time.

The import-secrets function imports SealedSecrets from a user's GitHub repository into the cluster, where the controller then attaches them as plain secrets.

### PR DESCRIPTION
**Pull request by heyts**
Signed-off-by: Thomas Schmidt <thomaschmidt@gmail.com>

## Description
This PR bumps kubernetes-client version to 1.15 for the `import-secrets` subproject. Kubernetes client v1.15 is the last client version supported by `github.com/bitnami-labs/sealed-secrets`.

Fixes #594

## How Has This Been Tested?
This PR was testing using `contrib/ci.sh`. The `ci.sh` script references `router/` which appears to have been renamed `edge-router/`. This PR doesn't update the script, but used the updated script using the `edge-router` directory to test. 

Tests are generally passing with the exception of one test returning an error as follow: 
```
- stat import-secrets/vendor/github.com/bitnami-labs/sealed-secrets/vendor_jsonnet/kube-libsonnet/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guest
book/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/gu
estbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples
/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examples/guestbook/lib/examp
les/guestbook/lib/examples/guestbook/lib: too many levels of symbolic links
```
This failure seems out-of-scope of the current PR.

## How are existing users impacted? What migration steps/scripts do we need?
No migration is necessary

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by heyts** [comment]
The build is now passing. I removed the offending symlinks by hand, however this may not be what we want. A better way would be to identify where the failure happens and restrict traversal of vendored files to exclude symlinks.

Being unfamiliar with the codebase, I do not know where the traversal takes place, but happy to make the change if someone points me to the correct location.

**Comment by alexellis** [comment]
Did you consider moving to Kubernetes 1.17?

**Comment by heyts** [comment]
I did consider it but it seems that `sealed-secrets` is still relying on a method (`DirectCodecFactory` specifically) that was removed from `apimachinery`  in 1.16. 

This PR documents the deprecation: https://github.com/kubernetes/apimachinery/commit/ed8af178d620cdb55580cddff6bfef9d3b00a7d7

**Comment by alexellis** [comment]
That doesn't mean that code has to be sunset at 1.15, it means we need to update the code.

https://github.com/inlets/inlets-operator
https://github.com/openfaas-incubator/ingress-operator 

I had to do this for one of these operators, take a look at the history / git-raw and you should find out what you need.

**Comment by alexellis** [comment]
https://github.com/inlets/inlets-operator/commit/7672519a7e08913377844a348f7e55b53c8a655a

**Comment by heyts** [comment]
Awesome, in this case I should be able to get all the way through to 1.17

**Comment by heyts** [comment]
@alexellis Building on the example you provided, I'm planning to extract `bitnami/sealed-secrets` to a `pkg/` directory inside of `import-secrets` to express that the code was changed, also updating the package imports.
This would allow me to update the outdated refs of `serializer.DirectCodecFactory`, in turn allowing to update `client-go` to 1.17.  Let me know how that sound and if there are any needed adjustments.

**Comment by alexellis** [comment]
Hi, could you update the description with

A) how this has been tested (end to end) - https://blog.alexellis.io/openfaas-cloud-for-development/
B) The 1.17 change (it still says 1.15)

Thanks,

Alex

File: import-secrets/handler.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func updateEncryptedData(ss *ssv1alpha1.SealedSecret, userSecret *SealedSecret)
 			return fmt.Errorf("can't decode base64 string (%s) - error: %s", k, err)
 		}
 
-		ss.Spec.EncryptedData[k] = encodedBytes
+		ss.Spec.EncryptedData[k] = string(encodedBytes)
```
Comment (alexellis) [inline import-secrets/handler.go]: Why did this change?

File: import-secrets/Gopkg.toml [hunk 2]
```diff
@@ -23,19 +23,19 @@
 
 [[constraint]]
   name = "k8s.io/client-go"
-  version = "7.0.0"
+  version = "0.15.0"
```
Comment (alexellis) [inline import-secrets/Gopkg.toml]: Were we not looking at 1.17?

**Comment by alexellis** [comment]
Thank you for this PR 👍 

I have tested e2e, but please do see my comments?

**Pull request merged**